app/src/androidTest/java/ServiceTests/StartStopService.java

```diff
@@ -54,6 +55,7 @@ public void verifyStartStopLocationServicesViaInitModule() {
         LogUITest.debug("Starting 'Save Location Service' & 'Ping Location Service Via Init Module'");
         LocationsHelper.INSTANCE.initLocationsModule(activityTestRule.getActivity().getApplication(), null, locationConfigs, locationPingServiceCallback, intent);
 
+        UiUtils.safeSleep(3);
```

Comment:
Can you do better than putting a blind sleep here

Reply:
Yes ... We are already using UIAutomator library .. We can use something from it .. But in all our GPS tests, We have added sleeps multiple times because we have to set frequency of SAVE service to like 2-3 seconds and each time we have to wait for `SAVE` service to get called for the next time to validate the data .

So, Just to not be inconsistent, I have added simple sleep .